XamlAnimatedGif.Shared/Animator.cs

```diff
@@ -553,6 +561,46 @@ private static Task<Stream> GetStreamFromUriAsync(Uri uri)
             }
             throw new NotSupportedException("Only ms-appx:, ms-appdata:, ms-resource: and file: URIs are supported");
         }
+
+        private static async Task<Stream> GetNetworkStreamAsync(Uri uri)
+        {
+            //generating temp file name by converting the uri to base64
+            var tempId = Convert.ToBase64String(Encoding.UTF8.GetBytes(uri.AbsoluteUri));
+
+            var folder = ApplicationData.Current.TemporaryFolder;
+
+            try
+            {
+                var tempFile = await folder.GetFileAsync(tempId);
+
+                //return the cache image
+                return await tempFile.OpenStreamForReadAsync();
+            }
+            catch{}
+
+            //no cache, continue with download
+            using (var client = new HttpClient())
+            {
+                var resp = await client.GetAsync(uri);
+
+                if (resp.IsSuccessStatusCode)
```

Comment:
It would be better to use `resp.EnsureSuccessStatusCode()` rather than checking the status code and manually throwing a generic exception